src/niMidi.cpp

```diff
@@ -205,18 +274,26 @@ class NiMidiSurface: public BaseSurface {
 	int _protocolVersion = 0;
 	int _bankStart = -1;
 
-	void _onBankChange() {
+	void _MixerUpdate() {
 		int numInBank = 0;
 		int bankEnd = this->_bankStart + BANK_NUM_TRACKS;
-		int numTracks = CSurf_NumTracks(false);
+		int numTracks = CSurf_NumTracks(false); // If we ever want to show just MCP tracks in KK Mixer View (param) must be (true)
 		if (bankEnd > numTracks) {
 			bankEnd = numTracks;
+			// Mark additional bank tracks as not available
+			int lastInBank = numTracks % BANK_NUM_TRACKS;
+			for (int i = 7; i > lastInBank; --i) {
+				this->_sendSysex(CMD_TRACK_AVAIL, 0, i);
```

Comment:
Ah. Interesting. So what happens with and without this visually? I assume this causes it to clear the mixer display for the old (now irrelevant) tracks?

Reply:
Exactly. In your master, when you enter the last bank with say just 3 tracks the display will still show 5 stale tracks to the right of those 3 tracks. Those 5 tracks belong to the preceding bank. They have to be explicitly "deleted".

Reply:
Also, the top row buttons are lit in white when a track is available. When you deactivate them, the lights go off, which is another handy cue.